modules/web/src/main/java/org/jboss/weld/module/web/context/beanstore/http/LazySessionBeanStore.java

```diff
@@ -85,9 +86,10 @@ protected HttpSession getSession(boolean create) {
         try {
             return SessionHolder.getSession(request, create);
         } catch (IllegalStateException e) {
-            // If container can't create an underlying session, invalidate the
-            // current one
+            // If container can't create an underlying session, invalidate the current one
             detach();
+            // re-throw the exception to properly show cause and message
+            Exceptions.rethrowException(e);
```

Comment:
I wonder if it wouldn't be safer (compatibility, etc.) to log an error instead?

Reply:
According to the issue (and the linked JBEAP issue, I figured having actual exception propagated would make more sense for clarity and debugging.